src/avahi_wrapper.cc

```diff
@@ -198,6 +198,29 @@ extern "C" void s_client_scan_callback(AvahiClient *client, AvahiClientState sta
     avahiWrapper->scanClientCallback(state);
 }
 
+// TBD: To put in a seperate file
+std::string str_tolower(const std::string s) {
+    std::string res(s);
+    std::transform(s.begin(), s.end(), res.begin(),
+                   [](unsigned char c){ return std::tolower(c); }
+                  );
```

Comment:
A more idiomatic way would be either:
```
std::string res;
res.reserve(s.size());
std::transform(s.begin(), s.end(), std::back_inserter(res),
  [](unsigned char c){ return std::tolower(c); }
);
```
or
```
std::string res(s);
std::transform(res.begin(), res.end(), res.begin(),
  [](unsigned char c){ return std::tolower(c); }
);
```
I'm not a fan of using naked output iterators (_i.e._ not wrapped with `std::back_inserter` or equivalent) with input interators from different collections, as it is very easy to forget to pre-allocate the output collection in that case and have a runtime crash because we're iteratively affecting values over an empty collection.